kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -261,23 +263,46 @@ def foundry_kompile(
             shutil.copy(req, req_path)
             regen = True
 
-    if regen or not foundry_main_file.exists():
-        requires = ['foundry.md']
+    _imports: dict[str, list[str]] = {contract.name: [] for contract in foundry.contracts.values()}
+    for i in imports:
+        imp = i.split(':')
+        if not len(imp) == 2:
+            raise ValueError(f'module imports must be of the form "[ContractName]:[MODULE-NAME]". Got: {i}')
+        if imp[0] in _imports:
+            _imports[imp[0]].append(imp[1])
+        else:
+            raise ValueError(f"Contract {imp[0]!r} doesn\'t exist.")
+
+    if regen or not foundry_contracts_file.exists() or not foundry_main_file.exists():
+        requires = []
         requires += [f'requires/{name}' for name in list(requires_paths.keys())]
-        imports = ['FOUNDRY'] + list(imports)
+        imports = ['FOUNDRY']
         kevm = KEVM(definition_dir)
         empty_config = kevm.definition.empty_config(Foundry.Sorts.FOUNDRY_CELL)
-        bin_runtime_definition = _foundry_to_bin_runtime(
+        bin_runtime_definition = _foundry_to_contract_def(
             empty_config=empty_config,
             contracts=foundry.contracts.values(),
+            requires=['foundry.md'],
+        )
+
+        contract_main_definition = _foundry_to_main_def(
             main_module=main_module,
-            requires=requires,
-            imports=imports,
+            empty_config=empty_config,
+            contracts=foundry.contracts.values(),
+            requires=(['contracts.k'] + requires),
+            imports=_imports,
+        )
+
+        kevm = KEVM(
+            definition_dir,
+            extra_unparsing_modules=(bin_runtime_definition.all_modules + contract_main_definition.all_modules),
         )
+        with open(foundry_contracts_file, 'w') as fcf:
+            _LOGGER.info(f'Writing file: {foundry_contracts_file}')
+            fcf.write(kevm.pretty_print(bin_runtime_definition) + '\n')
         with open(foundry_main_file, 'w') as fmf:
             _LOGGER.info(f'Writing file: {foundry_main_file}')
-            kevm = KEVM(definition_dir, extra_unparsing_modules=bin_runtime_definition.all_modules)
-            fmf.write(kevm.pretty_print(bin_runtime_definition) + '\n')
+            fmf.write(kevm.pretty_print(contract_main_definition) + '\n')
```

Comment:
We can probably say something simpler now:

```
foundry_contracts_file.write_text(kevm.pretty_print(bin_runtime_definition) + '\n')
_LOGGER.info(f'Wrote file: {foundry_contracts_file}')
foundry_main_file.write_text(kevm.pretty_print(contract_main_definition) + '\n')
_LOGGER.info(f'Wrote file: {foundry_main_file}')
```

Reply:
I made this change.